include/pressio/ops/tpetra_block/ops_rank1_update.hpp

```diff
@@ -145,13 +204,22 @@ update(T & v,         const a_Type &a,
        const T3 & v3, const d_Type &d,
        const T4 & v4, const e_Type &e)
 {
+  assert(::pressio::ops::extent(v, 0) == ::pressio::ops::extent(v1, 0));
+  assert(::pressio::ops::extent(v, 0) == ::pressio::ops::extent(v2, 0));
+  assert(::pressio::ops::extent(v, 0) == ::pressio::ops::extent(v3, 0));
+  assert(::pressio::ops::extent(v, 0) == ::pressio::ops::extent(v4, 0));
 
   using sc_t = typename ::pressio::Traits<T>::scalar_type;
   constexpr auto one  = ::pressio::utils::Constants<sc_t>::one();
-  v.update(b, v1, a); // v = a*v + b*v1
-  v.update(c, v2, one); // add c*v2
-  v.update(d, v3, one); // add d*v3
-  v.update(e, v4, one); // add e*v4
+  sc_t a_{a};
+  sc_t b_{b};
+  sc_t c_{c};
+  sc_t d_{d};
+  sc_t e_{e};
+  v.update(b_, v1, a_); // v = a*v + b*v1
+  v.update(c_, v2, one); // add c*v2
+  v.update(d_, v3, one); // add d*v3
+  v.update(e_, v4, one); // add e*v4
 }
 
 
```

Comment:
@fnrizzi should we keep the commented code below ?

Reply:
please remove it thanks!

Reply:
ok, thanks! fixed on 1403c478ca7a5b41af7637432de39e86a06ced1a